### PR TITLE
[docs] Add user stories to the hacking doc

### PR DESCRIPTION
**Pull request by sebsoto**

### CODE REVIEW

File: docs/HACKING.md [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+### I want to try a specific commit of WMCO, and deploy it using OLM
+* [Build the operator](#build)
+* [Deploy the operator through OLM](#deploying-the-operator-through-olm)
+
+### I want to deploy a specific commit of WMCO, in a way that is as close to the official deployment method as possible
+* [Build the operator](#build)
+* [Generate bundle manifests](#generating-new-bundle-manifests)
+* [Build the bundle image](#creating-a-bundle-image)
+* [Build an operator index](#creating-a-new-operator-index)
+* [Create a CatalogSource using the index](https://docs.openshift.com/container-platform/4.7/operators/admin/olm-managing-custom-catalogs.html#olm-creating-catalog-from-index_olm-managing-custom-catalogs)
```
Comment (saifshaikh48) [inline docs/HACKING.md:25]: Replacing the `4.7` in the URL with `latest` should keep us pointed to the most up-to-date official docs.

File: docs/HACKING.md [hunk 2]
```diff
@@ -1,6 +1,29 @@
 ⚠⚠⚠ THIS IS A LIVING DOCUMENT AND LIKELY TO CHANGE QUICKLY ⚠⚠⚠
 
 # Hacking on the WMCO
+Requires all pre-requisites in the [README](/README.md) to be fufilled.
+
+## User Workflows
+
+### I want to install an officially released version of WMCO on my OpenShift/OKD cluster
+* Instructions to install the community operator can be found in the [README](/README.md).
+  Paying customers should instead look at the [OpenShift docs.](https://docs.openshift.com/container-platform/latest/windows_containers/enabling-windows-container-workloads.html)
```
Comment (aravindhp) [inline docs/HACKING.md:10]: I suggest saying `Customers with a Red Hat subscription` instead of `Paying customers`

File: docs/HACKING.md [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+
+### I want to try a specific commit of WMCO
```
Comment (aravindhp) [inline docs/HACKING.md:12]: Shouldn't this be `I want to try the unreleased WMCO`? I really don't want to encourage folks trying a specific commit. Folks who are so inclined can do so at their own risk. Applies to all the headers.

File: docs/HACKING.md [hunk 4]
```diff
@@ -1,6 +1,29 @@
...
+* [Build the operator](#build)
+* [Deploy the operator directly](#deploying-the-operator-directly)
+
+### I want to try a specific commit of WMCO, and deploy it using OLM
+* [Build the operator](#build)
+* [Deploy the operator through OLM](#deploying-the-operator-through-olm)
+
+### I want to deploy a specific commit of WMCO, in a way that is as close to the official deployment method as possible
```
Comment (aravindhp) [inline docs/HACKING.md:20]: `I want to try the unreleased WMCO, and deploy it mimicking the official deployment method`

**Review by aravindhp — CHANGES_REQUESTED** [review]
I really like this approach, @sebsoto. Thanks for doing this.

File: docs/HACKING.md [hunk 5]
```diff
@@ -1,6 +1,29 @@
 ⚠⚠⚠ THIS IS A LIVING DOCUMENT AND LIKELY TO CHANGE QUICKLY ⚠⚠⚠
 
 # Hacking on the WMCO
+Requires all pre-requisites in the [README](/README.md) to be fufilled.
```
Comment (saifshaikh48) [inline docs/HACKING.md:4]: We might want to wait until #476 merges then link to the pre-reqs doc

Comment (aravindhp) [reply docs/HACKING.md:4]: Linking to the pre-req doc should be done in #476 wherever needed. IOW, this change can be removed from here but included in #476.

**Review by aravindhp — APPROVED** [review]
Thanks for doing this, @sebsoto 

/approve

**Pull request merged**